task_geo/data_sources/nasa/nasa.py

```diff
@@ -0,0 +1,214 @@
+    -------
+    df_nasa : pandas.DataFrame
+
+    """
+    # date column
+    df_nasa.reset_index(inplace=True, drop=False)
+    df_nasa.rename(columns={'index': 'date'}, inplace=True)
+    df_nasa['date'] = df_nasa['date'].apply(lambda dt:
+                                            pd.to_datetime(
+                                                f"{dt[:4]}-{dt[4:6]}-{dt[6:]}"
+                                            ).date())
+
+    # reorder columns
+    base_parms = ['country', 'region', 'sub_region', 'lon', 'lat', 'date']
+    all_parms = list(itertools.chain.from_iterable([PARAMETERS[p]
+                                                    for p in parms]))
+    df_nasa = df_nasa[base_parms + all_parms]
+
+    # rename columns
+    df_nasa.rename(columns=COL_NAMES, inplace=True)
+
+    return df_nasa
+
+
+def nasa_meteo_data(df, start_date, end_date=None,
+                    parms=["temperature", "humidity", "pressure"],
+                    level="region", join=False):
+    """
+    Retrieve meteorologic data from NASA.
+
+    Given a dataset with columns country, region, sub_region, lon, and lat, for
+    each geographic coordinate (lon, lat) corresponding to a place (specified
+    if country, region, or sub_region) extract the time series of the desired
+    data at the location.
+
+    Arguments:
+    ---------
+        df(pandas.DataFrame): Dataset with columns country, region, sub_region,
+                              lon, and lat
+        start_date(datetime64): Start date for the time series
+        end_date(pandas.Timestamp): End date fo rthe time series (optional)
+        parms(list of strings): Desired data, accepted are 'temperature',
+                                'humidity', and 'pressure' (optional)
+        level(string):  country, region, or sub_region, tells us at what
+                        granularity we want the data (eg for US sub_region, ie
+                        county, might make more sense than region, ie states,
+                        but in France region could be better, while for
+                        Luxembourg we probably want country) (optional)
+        join(bool): Determine if the meteorologic data has to be joined to the
+                    original dataset
+
+    Return:
+    ------
+        pandas.DataFrame:   Columns are country, region, sub_region (non-null),
+                            lon, lat, date, and the desired data.
+
+    """
+    # checks on the arguments
+    assert type(df) is pd.DataFrame, "Invalid input dataframe"
+    for col in ['country', 'region', 'sub_region', 'lon', 'lat']:
+        assert col in df.columns, f"Missing {col} column"
+    assert type(start_date) is pd.Timestamp, "Invalid start date"
+    if end_date:
+        assert type(end_date) is pd.Timestamp, "Invalid end date"
+    for p in parms:
+        assert p in PARAMETERS, f"Invalid parameter '{p}'"
+    assert level in ['country', 'region', 'sub_region'], \
+        "Invalid level {level}"
```

Comment:
Some remarks:
1. For our use case, this kind of defensive programming is not necessary. Let them pass wrong arguments and crash.
2. If you are gonna use it, dont check types with `type` use `isinstance`.
3. Make more clear messages, what fails and how to fix it.

Reply:
Ok, dropped this.